In [8]:
import pandas as pd
import glob
import cv2

# Background Dataset
I have collected a set of images that don't contain any waste, so their category will be `background (0)`.

In [9]:
first_set = glob.glob('../background-dataset/data/images/*')
second_set = glob.glob('../background-dataset/garbage-object-detection/images/*')
images_set = first_set + second_set

In [10]:
len(images_set) 

321

There are 321 background images in total.

## Dataset Creation

Let's create a dataframe with the same columns as the aggregated form of the annotations and images from COCO format.

In [11]:
columns = ['id', 'image_id', 'category_id', 'segmentation', 'area', 'bbox', 'iscrowd', 'filename', 'width', 'height']

In [16]:
data = []

for idx, image in enumerate(images_set):
    filename = image.split('/')[-1]

    img = cv2.imread(image)
    height, width, _ = img.shape
    
    info = (idx, idx, 0, [], 0., [], 0, filename, width, height)
    data.append(info)

In [17]:
background_df = pd.DataFrame(data, columns=columns)
background_df

id  image_id  category_id segmentation  area bbox  iscrowd  \
0      0         0            0           []   0.0   []        0   
1      1         1            0           []   0.0   []        0   
2      2         2            0           []   0.0   []        0   
3      3         3            0           []   0.0   []        0   
4      4         4            0           []   0.0   []        0   
..   ...       ...          ...          ...   ...  ...      ...   
316  316       316            0           []   0.0   []        0   
317  317       317            0           []   0.0   []        0   
318  318       318            0           []   0.0   []        0   
319  319       319            0           []   0.0   []        0   
320  320       320            0           []   0.0   []        0   

                                              filename  width  height  
0                 0cb80f833f1f49158c2b8854b0ba9f50.jpg   2048    1536  
1                                       veegvuil32.jpg   2048    1536  
2                                         00000038.jpg    500     281  
3                 71fdd0874142491bb86062b295086178.jpg   3264    2448  
4                 755e2c6967db4db782d522c525707909.jpg   3264    2448  
..                                                 ...    ...     ...  
316  mdc_photo_mdc_7c34f2_1511653787059_15116537953...   1944    1458  
317  mdc_photo_mdc_523960_1511628472142_15116284857...   1944    1458  
318  mdc_photo_mdc_5a256a_IMG_20170824_021706_15035...   4032    3024  
319          mdc_photo_cdv_photo_025_1511730688754.jpg   2448    2791  
320          mdc_photo_cdv_photo_096_1474742097585.jpg   1936    2592  

[321 rows x 10 columns]

In [20]:
background_df.to_csv('../background-dataset/background_df.csv', index=False)